joomla/joomla-cms
# [4.3][GUIDEDTOURS] Cleanup Tour model

Pull Request for Issue # .

### Summary of Changes
This PR makes some cleanup and code improvement to Tour model base on my code review:

- Use `$this->getCurrentUser()` to replace deprecated `Factory::getUser()`
- Fixed + simplify edit state check. The code `if (!$canEditState || !empty($item->default)) {` seems to be copied from somewhere, very likely from com_workflow code.
- Removed `$user->authorise('core.create', '__guidedtour_steps')` permission check. There is no asset `__guidedtour_steps`, so that check is useless
- Rename variable from $tour_id to $tourId to follow camel case naming convention.
- Some other small improvements

### Testing Instructions
1. Use Joomla 4.3 nightly build or any Joomla 4.3-dev installation has Guided Tours available
2. Apply patch, try to add/edit Tour and make sure nothing is broken

### Actual result BEFORE applying this Pull Request
Contains some bad/not optimized code


### Expected result AFTER applying this Pull Request
Cleaner code.

### Link to documentations
Please select:

- [z] No documentation changes for manual.joomla.org needed
